landlab/components/lateral_erosion/lateral_erosion.py

```diff
@@ -0,0 +1,631 @@
+            dt = min(abs(dtn), dt)
+            assert dt > 0.0, "timesteps less than 0."
+
+            # vol_lat is the total volume eroded from the lateral nodes through
+            # the entire model run. So vol_lat is itself plus vol_lat_dt (for current loop)
+            # times stable timestep size
+            vol_lat[:] += vol_lat_dt * dt
+            # this loop determines if enough lateral erosion has happened to change
+            # the height of the neighbor node.
+            for i in dwnst_nodes:
+                lat_node = lat_nodes[i]
+                wd = 0.4 * (da[i] * runoffms) ** 0.35
+                if (
+                    lat_node > 0
+                ):  # greater than zero now bc inactive neighbors are value -1
+                    if z[lat_node] > z[i]:
+                        # vol_diff is the volume that must be eroded from lat_node so that its
+                        # elevation is the same as node downstream of primary node
+                        # UC model: this would represent undercutting (the water height
+                        # at node i), slumping, and instant removal.
+                        if UC == 1:
+                            voldiff = (z[i] + wd - z[flowdirs[i]]) * grid.dx ** 2
+                        # TB model: entire lat node must be eroded before lateral
+                        # erosion occurs
+                        if TB == 1:
+                            voldiff = (z[lat_node] - z[flowdirs[i]]) * grid.dx ** 2
+                        # if the total volume eroded from lat_node is greater than the volume
+                        # needed to be removed to make node equal elevation,
+                        # then instantaneously remove this height from lat node. already
+                        # has timestep in it
+                        if vol_lat[lat_node] >= voldiff:
+                            dzlat[lat_node] = z[flowdirs[i]] - z[lat_node]  # -0.001
+                            # after the lateral node is eroded, reset its volume eroded
+                            # to zero
+                            vol_lat[lat_node] = 0.0
+
+            # multiply dzver(changed to dzdt above) by timestep size and combine with lateral erosion
+            # dzlat, which is already a length for the chosen time step
+            dz = dzdt * dt + dzlat
+            # change height of landscape
+            z[:] = dz + z
+            grid["node"]["topographic__elevation"][grid.core_nodes] = z[grid.core_nodes]
+            # update elapsed time
+            time = dt + time
+            # check to see that you are within 0.01% of the storm duration, if so
+            # done, if not continue
+
+            if time > 0.9999 * globdt:
+                time = globdt
+
+            else:
+                dt = globdt - time
+                qs_in = grid.zeros(centering="node")
+                # recalculate flow directions
+
+                fa = FlowAccumulator(
+                    grid,
```

Comment:
@alangston I would very very very strongly recommend not using the FlowAccumulator in this way. 

In stating that the typical outputs of the FlowAccumulator are required inputs to your component, you implicitly state an expectation that a user is going to run a FlowAccumulator independently of your component. But here, you create a new FlowAccumulator and run it. 

This isn't to say that you can't ensure that flow directions are recalculated as part of the adaptive method. 

If I were to want to do this, I'd pass the FlowAccumulator to the LateralEroder as an keyword argument.

Reply:
@kbarnhart, Thanks for the comments. They are all helpful and clear except for the comment above. Can you tell me a more about how to pass the FlowAccumulator to the LateralEroder as a keyword argument? Or point me to an example where this is done? Thanks!

Reply:
Thanks for the commits that implement passing the FlowAccumulator as a keyword. That change is now part of the LateralEroder.